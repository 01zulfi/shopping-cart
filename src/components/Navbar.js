import React from "react";
import { Link } from "react-router-dom";
import "../styles/Navbar.css";

const Navbar = ({ cartLength }) => {
  return (
    <div className="navbar">
      <h1>Asteroids</h1>
      <div className="nav-links">
        <ul>
          <li>
            <Link to="/home">Home</Link>
          </li>
          <li>
            <Link to="/shop">Shop</Link>
          </li>
          <li>
            <Link to="/shop/cart">Cart {cartLength}</Link>
          </li>
        </ul>
      </div>
    </div>
  );
};

export default Navbar;
