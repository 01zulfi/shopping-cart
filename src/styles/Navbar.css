div.navbar {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  background-color: rgba(255, 255, 255, 0.05);
  box-shadow: 0px 15px 30px rgb(0 0 0 / 15%);
  border-radius: 5px;
}

div.navbar h1 {
  margin: 0;
  margin-bottom: 0;
}

div.nav-links {
  width: 100%;
}

div.nav-links ul {
  padding: 0;
  width: inherit;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

div.nav-links > ul li {
  list-style: none;
}

div.nav-links > ul li a {
  text-decoration: none;
  color: white;
  font-size: 20px;
}

li.cart-link a {
  display: flex;
}

li.cart-link a p {
  margin-left: 5px;
  font-weight: bold;
}

@media (min-width: 768px) {
  div.navbar {
    flex-direction: row;
    justify-content: space-evenly;
    padding-top: 15px;
    padding-bottom: 15px;
  }

  div.nav-links {
    width: 15%;
  }

  div.nav-links ul {
    margin: 0;
    padding: 0;
    justify-content: space-evenly;
    margin-block-end: 0;
    width: 35%;
  }

  div.nav-links > ul li {
    margin-left: 35px;
  }
}
