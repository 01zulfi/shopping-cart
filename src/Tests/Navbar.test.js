import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import Navbar from "../components/Navbar";

test("navbar renders", () => {
  const { container } = render(
    <BrowserRouter>
      <Routes>
        <Route path="*" element={<Navbar />} />
      </Routes>
    </BrowserRouter>
  );
  expect(container.firstChild).toMatchSnapshot();
  expect(container.firstChild).toHaveClass("navbar");
});

test("navlink renders", () => {
  const { container, getByText } = render(
    <BrowserRouter>
      <Routes>
        <Route path="*" element={<Navbar />} />
      </Routes>
    </BrowserRouter>
  );
  const navlinks = container.querySelector(".nav-links");
  const home = getByText("Home");
  const shop = getByText("Shop");

  expect(navlinks).toBeInTheDocument();
  expect(home).toBeInTheDocument();
  expect(shop).toBeInTheDocument();
});
