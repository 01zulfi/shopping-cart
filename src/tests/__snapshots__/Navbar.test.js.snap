// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar renders 1`] = `
<div
  class="navbar"
>
  <h1>
    Asteroids
  </h1>
  <div
    class="nav-links"
  >
    <ul>
      <li>
        <a
          href="/home"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/shop"
        >
          Shop
        </a>
      </li>
      <li>
        <a
          href="/shop/cart"
        >
          Cart
        </a>
      </li>
    </ul>
  </div>
</div>
`;
